File: P2/P2.c
#include <stdio.h>

int main()
{
	int t1 = 1, t2 = 2, t3 = t1 + t2, sum = 2;

	while(t3 <= 4000000) {
		if(t3 % 2 == 0)
			sum += t3;
		t1 = t2;
		t2 = t3;
		t3 = t1 + t2;
	}

	printf("%d\n", sum);

	return 0;
}

File: problem2/P2.rb
sum = 0
fib1 = 1
fib2 = 2
tfib = fib2

while tfib <= 4000000
	sum += tfib if tfib % 2 == 0

	tfib = fib1 + fib2
	fib1 = fib2
	fib2 = tfib
end

puts sum
